sdk/trace/trace_test.go

```diff
@@ -1972,15 +1972,91 @@ func TestWithIDGenerator(t *testing.T) {
 
 			gotSpanID, err := strconv.ParseUint(span.SpanContext().SpanID().String(), 16, 64)
 			require.NoError(t, err)
-			assert.Equal(t, uint64(startSpanID+i), gotSpanID)
+			assert.Equal(t, uint64(startSpanID)+uint64(i), gotSpanID)
 
 			gotTraceID, err := strconv.ParseUint(span.SpanContext().TraceID().String(), 16, 64)
```

Comment:
Unrelated to these changes, but we should be testing both the high and low 64-bits of the trace ID here.

Reply:
https://github.com/open-telemetry/opentelemetry-go/issues/7160